src/coreclr/debug/di/rsappdomain.cpp

```diff
@@ -834,16 +834,18 @@ CordbModule* CordbAppDomain::LookupOrCreateModule(VMPTR_Module vmModule, VMPTR_D
 
     _ASSERTE(!vmDomainAssembly.IsNull() || !vmModule.IsNull());
 
+    if (vmModule.IsNull())
+        GetProcess()->GetDAC()->GetModuleForDomainAssembly(vmDomainAssembly, &vmModule);
+
+    _ASSERTE(!vmModule.IsNull());
+
     // check to see if the module is present in this app domain
-    pModule = m_modules.GetBase(vmDomainAssembly.IsNull() ? VmPtrToCookie(vmModule) : VmPtrToCookie(vmDomainAssembly));
+    pModule = m_modules.GetBase(VmPtrToCookie(vmModule));
     if (pModule != NULL)
     {
         return pModule;
     }
 
-    if (vmModule.IsNull())
-        GetProcess()->GetDAC()->GetModuleForDomainAssembly(vmDomainAssembly, &vmModule);
-
     RSInitHolder<CordbModule> pModuleInit(new CordbModule(GetProcess(), vmModule, vmDomainAssembly));
```

Comment:
I assume in order to hit the bug previously we had one code path calling into this function where vmModule=NULL and a 2nd calling in where vmDomainAssembly=NULL. If LookupOrCreateModule(non-NULL domainAssembly, NULL module) now returns a module object where m_domainAssembly=NULL, various ICorDebugModule APIs that rely on m_domainAssembly won't work. I suspect you either need to ensure that m_domainAssembly is never NULL, or you need to change the code in CordbModule to no longer depend on it.

Reply:
My observation was that DB_IPCE_LOAD_MODULE provides the domain assembly and is received first for each module before we attempt any lookups on it. But I am not confident this will always be the case, so I'll add a check and look up the domain assembly if needed.